main.go

```diff
@@ -423,7 +423,10 @@ func runTests(perfStatsForTest *perfTestUtils.PerfStats, mode int, testSuite *te
 		}
 	}
 
-	time.Sleep(time.Second * 1)
+	// Kill the peak memory thread to avoid race condition when saving metrics.
+	chanQuitPkMem <- true
+
```

Comment:
Signal the peak memory goroutine to exit. With the thread complete, we can now use the data structures below without mutex locking.